tests/robot-tests/scripts/visual-testing/generate_tables_and_charts_test_cases.py

```diff
@@ -0,0 +1,53 @@
+import csv
+from logging import warning
+import argparse
+import os
+import shutil
+
+ap = argparse.ArgumentParser()
+ap.add_argument("-f", "--file", required=True,
+                help="the datablocks csv to use to generate the robot test script from the template")
+ap.add_argument("-t", "--target", required=True,
+                help="the target filename of the generated robot test script")
+args = vars(ap.parse_args())
+
+datablocks_csv_filepath = args['file']
+
+release_urls = set()
+
+with open(datablocks_csv_filepath, 'r', encoding='utf-8-sig') as csv_file:
+    csv_reader = csv.reader(csv_file, delimiter=',')
+    next(csv_reader)
+    for row in csv_reader:
+        publication_slug = row[4]
+        release_slug = row[2]
+        release_urls.add(f'/find-statistics/{publication_slug}/{release_slug}')
+
+datablocks_csv_filename = os.path.basename(datablocks_csv_filepath)
+
+robot_tests_folder = '../../tests/visual_testing'
+
+with open(f'{robot_tests_folder}/visually_check_tables_and_charts.template.robot', 'r', encoding='utf-8-sig') as robot_template_file:
+
+    test_template = robot_template_file.read()
+
+    header, template_and_footer = test_template.split('{{test_case_template}}')
+    test_case_template_section, footer = template_and_footer.split('{{/test_case_template}}')
+    generated_test_script = header
+
+    for release_url in release_urls:
+        warning(release_url)
+        generated_test_script += test_case_template_section.replace('{{release_url}}', release_url)
+
+    target_robot_test_filename = args['target']
+    target_robot_test_csv_filename = f'{target_robot_test_filename}.csv'
+
+    generated_test_script += footer.replace('{{datablocks_csv_filename}}',
+                                            f'tests/visual_testing/{target_robot_test_csv_filename}')
+
+    warning(generated_test_script)
```

Comment:
Do we need this warning or should be log something such as `warning("generated test script")`

Reply:
It was actually just so that we could see the whole generated script in the console when it has been generated to get easy eyes on it.  Happy to remove?

Reply:
Nah I think that's ok. That'll be useful for a quick sanity check to the script has been generated as expected